cider/tests/Sql2IR.cpp

```diff
@@ -0,0 +1,158 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <cstddef>
+#include <iostream>
+
+#include <google/protobuf/util/json_util.h>
+#include <boost/program_options.hpp>
+
+#include "cider/CiderAllocator.h"
+#include "cider/CiderCompileModule.h"
+#include "cider/CiderOptions.h"
+#include "cider/CiderTableSchema.h"
+#include "cider/batch/CiderBatch.h"
+#include "substrait/plan.pb.h"
+#include "substrait/type.pb.h"
+#include "util/Logger.h"
+#include "utils/Utils.h"
+
+const substrait::Rel* findJoinRightNode(const substrait::Rel& rel_node) {
+  const substrait::Rel::RelTypeCase& rel_type = rel_node.rel_type_case();
+  switch (rel_type) {
+    case substrait::Rel::RelTypeCase::kJoin:
+      return &rel_node.join().right();
+    case substrait::Rel::RelTypeCase::kFilter:
+      return findJoinRightNode(rel_node.filter().input());
+    case substrait::Rel::RelTypeCase::kProject:
+      return findJoinRightNode(rel_node.project().input());
+    case substrait::Rel::RelTypeCase::kAggregate:
+      return findJoinRightNode(rel_node.aggregate().input());
+    default:
+      return nullptr;
+  }
+}
+
+const substrait::ReadRel* findReadNode(const substrait::Rel& rel_node) {
+  const substrait::Rel::RelTypeCase& rel_type = rel_node.rel_type_case();
+  switch (rel_type) {
+    case substrait::Rel::RelTypeCase::kRead:
+      return &rel_node.read();
+    case substrait::Rel::RelTypeCase::kFilter:
+      return findReadNode(rel_node.filter().input());
+    case substrait::Rel::RelTypeCase::kProject:
+      return findReadNode(rel_node.project().input());
+    case substrait::Rel::RelTypeCase::kAggregate:
+      return findReadNode(rel_node.aggregate().input());
+    default:
+      return nullptr;
+  }
+}
+
+void feedBuildTable(substrait::Plan& plan,
+                    std::shared_ptr<CiderCompileModule> cider_compile_module) {
+  substrait::Rel root = plan.relations(0).root().input();
+  const substrait::Rel* right_node = findJoinRightNode(root);
+  if (right_node == nullptr) {
+    return;
+  }
+  const substrait::ReadRel* read_node = findReadNode(*right_node);
+  auto& schema = read_node->base_schema();
+  size_t count = schema.names_size();
+
+  std::vector<std::string> col_names;
+  std::vector<::substrait::Type> col_types;
+  for (int i = 0; i < count; ++i) {
+    col_names.emplace_back(schema.names(i));
+    col_types.emplace_back(schema.struct_().types(i));
+  }
+
+  auto table_schema = std::make_shared<CiderTableSchema>(col_names, col_types);
+
+  cider_compile_module->feedBuildTable(CiderBatch(1, table_schema));
+}
+
+int main(int argc, char** argv) {
```

Comment:
Better to add a tutorial in https://intel.github.io/BDTK/develop.html and make other aware of this great tool.